### PR TITLE
feat: gitlab-11.2-group-api-improvements

### LINKED ISSUE #5: Info: Owned groups vs Master access
I would like to know why your intention of only limiting it to owned groups vs groups a user has write access to? 

Seems like something that doesn't really match up. In most cases, Master access seems to make more sense.

Is this due to a limitation on the gitlab api API?  https://docs.gitlab.com/ce/api/groups.html#list-groups

### PR DESCRIPTION
**Pull request by dlouzan**
- use the new gitlab 11.2 `min_access_level` parameter in the groups api:
  publish rights are now configurable based on the access level of the
  user to the groups, it doesn't require owner permissions
- add new configuration parameters for `access` and `publish` access
  levels

Fixes #5

### CODE REVIEW

**Comment by dlouzan** [comment]
Quick question: if we are going to add the legacy flag in the configuration, which will be disabled by default, I guess we can also set the default `publish` access level to `$maintainer`? At least it feels more natural to me. When `legacy_mode: true`, the value applied will be `$owner` anyway, there's no other option pre-11.2.

**Comment by bufferoverflow** [comment]
Good idea @dlouzan , fully agree!

**Comment by dlouzan** [comment]
Just pushed the latest changes:

- remove `access` access level configuration parameter since it cannot be used with the current plugins api
- support gitlab 9.0+ with a new `legacy_mode` configuration flag; when not in legacy mode, gitlab 11.2+ is required
- improved README, document all configuration options

File: conf/localhost.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+storage: /tmp/storage/data
```
[the remark this answers is not in the document]
Comment (dlouzan) [reply conf/localhost.yaml]: Ooops, that's for my local testing, shouldn't be there

File: README.md [hunk 2]
```diff
@@ -8,16 +8,20 @@ as authentication provider for the private npm registry
...
 - user authenticates with Personal Access Token
-- owned groups (no subgroups) are added to the user
-- publish packages if package scope or name is an owned group name
+- access & publish packages depending on user rights in gitlab
 
 > This is experimental!
 
+## Gitlab Version Compatibility
+
+- If `legacy_mode: false`: Gitlab 11.2+
```
[the remark this answers is not in the document]
Comment (dlouzan) [reply README.md]: It's indicated below in the configuration options list that `legacy_mode: false` is the default, but I will make it also explicit here.

**Comment by bufferoverflow** [comment]
Great work!

**Pull request merged**